src/repr/src/adt/interval.rs

```diff
@@ -107,18 +115,17 @@ impl Interval {
             return None;
         }
 
-        let seconds =
-            self.dur_as_secs() as f64 * other + months.fract() * 60.0 * 60.0 * 24.0 * 30.0;
-        if seconds.is_nan() || seconds < i64::MIN as f64 || seconds > i64::MAX as f64 {
+        let days = self.days as f64 * other + months.fract() * 30.0;
```

Comment:
This function already did all this and you're just changing it but: there's lots of unchecked multiplication, addition, and `as` casting in this function. Can you either check all those or write comments describing why they are guaranteed to be safe?

Reply:
I refactored this to remove as much of the `as` as I could. As far as I can tell, the only way to do a lossy conversion between `f64` and `i64`/`i32` is with `as` so I had to leave a couple of them in. If there's a better way please let me know because I couldn't figure it out.

There's no check arithmetic because it's all floating point arithmetic which instead of overflowing/underflowing will go to +inf/-inf. I've added checks for those and returned `None` in those cases.